File: swift/Sources/threads/Commands/GitCmd.swift
import ArgumentParser
import Foundation

struct GitCmd: ParsableCommand {
    static var configuration = CommandConfiguration(
        commandName: "git",
        abstract: "Show pending thread changes"
    )

    func run() throws {
        let ws = try getWorkspace()

        let threads = try findAllThreads(ws)

        var modified: [String] = []
        for t in threads {
            let relPath = t.relativePath(from: ws) ?? t
            if gitHasChanges(ws, relPath) {
                modified.append(relPath)
            }
        }

        if modified.isEmpty {
            print("No pending thread changes.")
            return
        }

        print("Pending thread changes:")
        for f in modified {
            print("  \(f)")
        }
        print()
        print("Suggested:")
        print("  git add \(modified.joined(separator: " ")) && git commit -m \"threads: update\" && git push")
    }
}

File: rust/src/cmd/git_cmd.rs
use std::path::Path;

use clap::Args;

use crate::git;
use crate::workspace;

#[derive(Args)]
pub struct GitArgs {}

pub fn run(_args: GitArgs, ws: &Path) -> Result<(), String> {
    let threads = workspace::find_all_threads(ws)?;

    let mut modified = Vec::new();
    for t in threads {
        let rel_path = t
            .strip_prefix(ws)
            .map(|p| p.to_string_lossy().to_string())
            .unwrap_or_else(|_| t.to_string_lossy().to_string());

        if git::has_changes(ws, &rel_path) {
            modified.push(rel_path);
        }
    }

    if modified.is_empty() {
        println!("No pending thread changes.");
        return Ok(());
    }

    println!("Pending thread changes:");
    for f in &modified {
        println!("  {}", f);
    }
    println!();
    println!("Suggested:");
    println!(
        "  git add {} && git commit -m \"threads: update\" && git push",
        modified.join(" ")
    );

    Ok(())
}
